plugins/goodix-tp/fu-goodixtp-hid-device.c

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (C) 2023 Goodix.inc <xulinkun@goodix.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#ifdef HAVE_HIDRAW_H
+#include <linux/hidraw.h>
+#include <linux/input.h>
+#endif
+
+#include "fu-goodixtp-common.h"
+#include "fu-goodixtp-firmware.h"
+#include "fu-goodixtp-hid-device.h"
+
+typedef struct {
+	guint8 patch_pid[9];
+	guint8 patch_vid[4];
```

Comment:
Question: what do we actually need the patch_pid and patch_vid _for_?

Reply:
Yes, patch_pid storage chip name such as GT7863, GT9966.... and vid storage firmware version. We need to print them.